### DISCUSSION TITLE
One library of five takes hours to complete [Docker]

### DISCUSSION THREAD

**Discussion by marcb199** [meisnate12/Plex-Meta-Manager#1291] [Q&A]
I have configured PMM to apply overlays to 5 of my Plex libraries (3 movie libraries, 2 tv show libraries). PMM works pretty much perfectly fine on all but one movie library. It is the largest of the movie libraries with 560 movies. PMM applied all the overlays on the first run perfectly fine, but the second and any subsequent run takes roughly 6 hours to complete. Looking into the logs, pretty much for every movie the following error occurs: 

[ERROR]    | Overlays Attempted on (...)

At first I thought this was expected due to the size of the library, but one of my tv show libraries with approximately 6.000 episodes finishes within 10 - 15 minutes. It seems like PMM is not "aware" that it has already applied the overlays to the movie posters. 

Any idea on how to fix this?

Thanks

**Comment by chazlarson** [comment]
Do you have `reapply_overlays` set to true?  If so, you're telling PMM to reapply overlays on every run regardless if it thinks it needs to.

**Reply by marcb199** [reply]
Hi, no, I didn't even include this attribute in my config.yml. For the overlays it looks like this:

overlay_path:
    - remove_overlays: false
    - pmm: resolution
    - pmm: audio_codec

**Comment by chazlarson** [comment]
Can you provide a log of this happening?

**Reply by marcb199** [reply]
Sure, here is the log:

https://code.mcloudx.com/MyAuquRs

Thanks for your help!

**Reply by marcb199** [reply]
Update:
So I got it running once with almost no errors, but in the end I gave up on running PMM with Docker. This might be a limitation of Docker for Mac, but PMM wouldn't run reliably. The process inside the container would suddenly stop or wouldn't process all libraries in one run (only one library at a time and even then, most of the time, it wouldn't work). Funny thing is, the only other container that also wouldn't run reliably with Docker for Mac and that I had to move to my Raspberry Pi was Plex Auto Language. 
I'm now running PMM with the local installation method and this works. At least 2 test runs went through without a hitch. 
I guess this can be closed.
